org.uqbar.project.wollok.lib/src/wollok.wlk

```diff
@@ -45,7 +75,249 @@ package lang {
+		
+		method map(closure) = this.fold(this.newInstance(), [ acc, e |
+			 acc.add(closure.apply(e))
+			 acc
+		])
+		
+		method filter(closure) = this.fold(this.newInstance(), [ acc, e |
+			 if (closure.apply(e))
+			 	acc.add(e)
+			 acc
+		])
+				
+		method contains(e) = this.exists[one | e == one ]
+		
+		override method internalToSmartString(alreadyShown) {
+			return this.toStringPrefix() + this.map[e| e.toSmartString(alreadyShown) ].join(', ') + this.toStringSufix()
+		}
+		
+		method toStringPrefix()
+		method toStringSufix()
+		
+		method newInstance()
+	}
+
+	/**
+	 *
+	 * @author jfernandes
+	 * @since 1.3
+	 */	
+	class Set extends Collection {
+	
+		override method newInstance() = new Set()
+		override method toStringPrefix() = "#{"
+		override method toStringSufix() = "}"
+		
+		method any() = this.first()
+		
+		method first() native
+		
+		// REFACTORME: DUP METHODS
+		method fold(initialValue, closure) native
+		method add(element) native
+		method remove(element) native
+		method size() native
+		method clear() native
+		method join(separator) native
+		method join() native
+		method equals(other) native
+		method ==(other) native
+	}
+	
+	/**
+	 *
+	 * @author jfernandes
+	 * @since 1.3
+	 */
+	class List extends Collection {
+
+		method get(index) native
+		
+		override method newInstance() = new List()
+		
+		method any() {
+			if (this.isEmpty()) 
+				throw new Exception() //("Illegal operation 'any' on empty collection")
+			else 
+				return this.get(this.randomBetween(0, this.size()))
+		}
+		
+		override method toStringPrefix() = "#["
+		override method toStringSufix() = "]"
+		
+		// REFACTORME: DUP METHODS
+		method fold(initialValue, closure) native
+		method add(element) native
+		method remove(element) native
+		method size() native
+		method clear() native
+		method join(separator) native
+		method join() native
+		method equals(other) native
+		method ==(other) native
+	}
+	
+	/**
+	 *
+	 * @author jfernandes
+	 * @since 1.3
+	 * @noInstantiate
+	 */	
+	class Number {
+		method max(other) = if (this >= other) this else other
+		method min(other) = if (this <= other) this else other
+		
+		method !=(other) = ! (this == other)
+	}
+	
+	/**
+	 * @author jfernandes
+	 * @since 1.3
+	 * @noInstantiate
+	 */
+	class Integer extends Number {
+		method ==(other) native
+		method +(other) native
+		method -(other) native
+		method *(other) native
+		method /(other) native
+		method **(other) native
+		method %(other) native
+		
+		method toString() native
+		
+		override method internalToSmartString(alreadyShown) { return this.stringValue() }
+		method stringValue() native	
+		
+		method ..(end) = new Range(this, end)
+		
+		method >(other) native
+		method >=(other) native
+		method <(other) native
+		method <=(other) native
+		
+		method abs() native
+		method invert() native
+	}
+	
+	/**
+	 * @author jfernandes
+	 * @since 1.3
+	 * @noInstantiate
```

Comment:
Why is this annotation only in Integer/Number and not in other primitive types (e.g. Boolean)